python/lsst/daf/butler/remote_butler/server/handlers/_external.py

```diff
@@ -140,33 +138,57 @@ def find_dataset(
 ) -> SerializedDatasetRef | None:
     collection_query = query.collections if query.collections else None
 
-    # Get the simple dict from the SerializedDataCoordinate. We do not know
-    # if it is a well-defined DataCoordinate or needs some massaging first.
-    # find_dataset will use dimension record queries if necessary.
-    data_id = query.data_id.dataId
-
     butler = factory.create_butler()
     ref = butler.find_dataset(
         dataset_type,
-        None,
+        query.data_id,
         collections=collection_query,
-        storage_class=query.storage_class,
         timespan=None,
         dimension_records=query.dimension_records,
         datastore_records=query.datastore_records,
-        **data_id,
     )
     return ref.to_simple() if ref else None
 
 
-@external_router.post("/v1/get_file")
+@external_router.get(
+    "/v1/get_file/{dataset_id}",
+    summary="Lookup via DatasetId (UUID) the information needed to download and use the files associated"
+    " with a dataset.",
+)
 def get_file(
-    request: GetFileRequestModel,
+    dataset_id: uuid.UUID,
     factory: Factory = Depends(factory_dependency),
 ) -> GetFileResponseModel:
     butler = factory.create_butler()
-    ref = butler.get_dataset(request.dataset_id, datastore_records=True)
+    ref = butler.get_dataset(dataset_id, datastore_records=True)
     if ref is None:
-        raise NotFoundException(f"Dataset ID {request.dataset_id} not found")
+        raise NotFoundException(f"Dataset ID {dataset_id} not found")
+    return _get_file_by_ref(butler, ref)
+
+
+@external_router.post(
+    "/v1/get_file_by_data_id",
+    summary="Lookup via DataId (metadata key/value pairs) the information needed to download"
+    " and use the files associated with a dataset.",
+)
+def get_file_by_data_id(
+    request: GetFileByDataIdRequestModel,
+    factory: Factory = Depends(factory_dependency),
+) -> GetFileResponseModel:
+    butler = factory.create_butler()
+    try:
+        ref = butler._findDatasetRef(
+            datasetRefOrType=request.dataset_type_name,
+            dataId=request.data_id,
+            collections=request.collections,
+            datastore_records=True,
+        )
+        return _get_file_by_ref(butler, ref)
+    except LookupError as e:
+        raise NotFoundException() from e
```

Comment:
No error message?

Reply:
It's not actually shipping the error message from those NotFoundExceptions to the client yet, and the `from e` will put the original error message in the server logs.  I suppose I could have put in "DataId not found" or something.

In a couple weeks when I go in and set up the error propagation framework, that try block will go away entirely.